lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -647,7 +647,20 @@ double CGHeroInstance::getFightingStrength() const
 
 double CGHeroInstance::getMagicStrength() const
 {
-	return sqrt((1.0 + 0.05*getPrimSkillLevel(PrimarySkill::KNOWLEDGE)) * (1.0 + 0.05*getPrimSkillLevel(PrimarySkill::SPELL_POWER)));
+	if (!hasSpellbook())
+		return 1;
+	bool atLeastOneCombatSpell = false;
+	for (auto spell : spells)
+	{
+		if (spellbookContainsSpell(spell) && spell.toSpell()->isCombat())
+		{
+			atLeastOneCombatSpell = true;
+			break;
+		}
+	}
+	if (!atLeastOneCombatSpell)
+		return 1;
+	return sqrt((1.0 + 0.05*getPrimSkillLevel(PrimarySkill::KNOWLEDGE) * mana / manaLimit()) * (1.0 + 0.05*getPrimSkillLevel(PrimarySkill::SPELL_POWER) * mana / manaLimit()));
```

Comment:
This will also affect selection of hero to transfer in campaigns

Reply:
Can we add a default-parameter like "bool useMaxMana = false", set it to true for the selection of heroes to transfer in campaigns and if it's true use it as if mana is manaLimit?

Reply:
Made a separate version of getMagicStrengthForCampaign and getHeroStrengthForCampaign, so that heroes' magic-power is calculated the traditional way for the purpose of whether the hero should be brought for the next campaign-mission.